kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -179,11 +179,20 @@ simplifyClaimRule =
     worker :: ClaimPattern -> LogicT simplifier ClaimPattern
     worker claimPattern = do
         let lhs = ClaimPattern.left claimPattern
-        simplified <- simplify lhs
-        let substitution = Pattern.substitution simplified
-            lhs' = simplified { Pattern.substitution = mempty }
+            rhs = ClaimPattern.right claimPattern
+        simplifiedLhs <- simplify lhs
+        simplifiedRhs <- OrPattern.filterOr <$> traverse simplify rhs
+        let substitution =
+                Pattern.substitution simplifiedLhs
+                <> foldMap Pattern.substitution (MultiOr.getMultiOr simplifiedRhs)
```

Comment:
We cannot add the substitution from the right-hand side to the global substitution. We can't apply it to the left-hand side, and it isn't right to drop it from the right-hand side (below) either.

Reply:
Initially I didn't do that, but I tried it out of other ideas

Reply:
Initially I didn't do that, but I tried it out of other ideas